tools/social-twitter/README.md

```diff
@@ -1106,6 +1106,108 @@ It's important to note that some errors may have either a specific error kind (l
 
 ---
 
+# `xyz.taluslabs.social.twitter.upload-media@1`
+
+Standard Nexus Tool that uploads media to Twitter.
+Twitter api [reference](https://docs.x.com/x-api/media/media-upload)
+
+## Input
+
+**Authentication Parameters**
+
+The following authentication parameters are provided as part of the TwitterAuth structure:
+
+- **`consumer_key`: [`String`]** - Twitter API application's Consumer Key
+- **`consumer_secret_key`: [`String`]** - Twitter API application's Consumer Secret Key
+- **`access_token`: [`String`]** - Access Token for user's Twitter account
+- **`access_token_secret`: [`String`]** - Access Token Secret for user's Twitter account
+
+**Additional Parameters**
+
+**`media_data`: [`String`]**
+
+The Base64 encoded media content to upload.
+
+**`media_type`: [`MediaType`]**
+
+The MIME type of the media being uploaded. This is an enum type that includes:
+
+- `ImageJpeg` - For JPEG images
+- `ImageGif` - For GIF images
+- `ImagePng` - For PNG images
+- `ImageWebp` - For WebP images
+- `VideoMp4` - For MP4 videos
+- `VideoWebm` - For WebM videos
+- `VideoMp2t` - For MPEG-2 Transport Stream videos
```

Comment:
these should be changed to the values that are passed in the request so `image/jpeg` and similar